install/kubernetes/Makefile

```diff
@@ -61,7 +61,6 @@ update-versions:
 			branch="$$cilium_version";										\
 			hubble_version="$(HUBBLE_UI_VERSION)";									\
 			hubble_backend_version="$(HUBBLE_UI_BACKEND_VERSION)";							\
-			pull_policy="Always";											\
```

Comment:
I think this is deliberate, to ensure that when developers push new versions of the image with the same tag, then restart the pods, then the cluster will pick up the new release. Otherwise developers have to take additional action to force the new image to be pulled.

For the actual releases that we expect Cilium users to use, the default should already be `IfNotPresent`.

Reply:
I second what @joestringer said.

@TrevorTaoARM I can see that you are thinking somewhat differently of this, as you mentioned these images built by users. It sounds like you have slightly different context to this, could you please elaborate a little, so we can get everyone on the same page?

Reply:
@joestringer @errordeveloper 
Just as you said, I think it's really interesting why we set "pull_policy="IfNotPresent" in line 50, then we set 'pull_policy="Always";' in line 57 and 64 separately, so what about the use of the first setting here?
Can we just set the 'pull_policy="Always"' in line 50?

Reply:
The condition on line 52 asks the question "Is the version a standard development version on master (`x.y.90`)?". This is used when generating helm charts in the master branch. The condition on line 59 asks "Is the version a development version targeted for a specific branch (`x.y-dev`)? This is used when generating development versions of the charts for stable branches (see the bottom of the list at https://github.com/cilium/charts ).

The setting on line 50 is used for all other cases: Real, stable releases where we do not want to force users in production environments to inadvertently pull images when they do not intend to do so.

Reply:
For `x.y-dev` charts, the image is not stable and will change over time, so if we set this to `IfNotPresent` then developers could end up with a random old development version based on a branch at some point in history. Given that they are also developing Cilium, I would expect that they intend to pull new versions of the images all the time and they're likely to hit more issues if the image pull policy is set to avoid pulling newer image versions.